src/app/utils/firestore.ts

```diff
@@ -0,0 +1,62 @@
+import { Course, Question } from "../types/db";
```

Comment:
Side note, I think the converter should be in a dedicated folder. Maybe `services/firestore`?